src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -1659,11 +2115,793 @@ End Class
+                //     public override string P => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "P").WithArguments("covariant returns").WithLocation(12, 28)
+                );
+            verify(comp);
+            comp = CreateCompilationWithCovariantReturns(source).VerifyDiagnostics(
+                );
+            verify(comp);
+            verify(CompilationReferenceView(comp));
+            verify(MetadataView(comp));
+            verify(RetargetedView(comp));
+
+            static void verify(CSharpCompilation comp)
+            {
+                VerifyNoOverride(comp, "A.P");
+                VerifyNoOverride(comp, "A.get_P");
+                VerifyNoOverride(comp, "B.P");
+                VerifyNoOverride(comp, "B.get_P");
+                VerifyOverride(comp, "C.P", "System.String C.P { get; }", "System.Object B.P { get; }");
+                VerifyOverride(comp, "C.get_P", "System.String C.P.get", "System.Object B.P.get");
+            }
+        }
+
+        [Fact]
+        public void OverlappingMethodimplRequirements_03()
+        {
+            var source = @"
+public class A
+{
+    public virtual string M() => null;
+}
+public class B : A
+{
+    public virtual new object M() => null;
+}
+public class C : B
+{
+    public override string M() => null;
+}
+";
+            var comp = CreateCompilationWithoutCovariantReturns(source).VerifyDiagnostics(
+                // (12,28): error CS8652: The feature 'covariant returns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public override string M() => null;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "M").WithArguments("covariant returns").WithLocation(12, 28)
+                );
+            verify(comp);
+            comp = CreateCompilationWithCovariantReturns(source).VerifyDiagnostics(
+                );
+            verify(comp);
+            verify(CompilationReferenceView(comp));
+            verify(MetadataView(comp));
+            verify(RetargetedView(comp));
+
+            static void verify(CSharpCompilation comp)
+            {
+                VerifyNoOverride(comp, "A.M");
+                VerifyNoOverride(comp, "B.M");
+                VerifyOverride(comp, "C.M", "System.String C.M()", "System.Object B.M()");
+            }
+        }
+
+        [Fact]
+        public void InExpressionTree_01()
+        {
+            var source = @"
+using System;
+using System.Linq.Expressions;
+public class Base
+{
+    public virtual object M() => null;
+    public virtual object P => null;
+}
+public class Derived : Base
+{
+    public override string M() => null;
+    public override string P => null;
+}
+public class Program : Derived
+{
+    Expression<Func<Derived, string>> M1()
+    {
+        return d => d.M();
```

Comment:
>d [](start = 20, length = 1)

Consider also testing with ```base``` as receiver.

Reply:
That is a semantic error.

---
In reply to: [424787359](https://github.com/dotnet/roslyn/pull/44025#discussion_r424787359) [](ancestors = 424787359)